backend/howtheyvote/pipelines/common.py

```diff
@@ -1,6 +1,68 @@
+import hashlib
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any
+
+from requests import Response
+from structlog import get_logger
+
+from ..models import PipelineStatus
+from ..scrapers import ScrapingError
+
+log = get_logger(__name__)
+
+
+@dataclass
+class PipelineResult:
+    status: PipelineStatus
+    checksum: str | None
+
+
 class PipelineError(Exception):
     pass
 
 
 class DataUnavailableError(PipelineError):
     pass
+
+
+class DataUnchangedError(PipelineError):
+    pass
```

Comment:
Note to self: These aren’t actually errors, they are exceptions used for flow control. I think I’ve blindly followed the linter recommendation, but this naming doesn’t really make sense for this case.

According to PEP 8, exception classes should have an `Error` suffix, but only if they actually are errors, otherwise they should have no special suffix, so in this case `DataUnchanged` and `DataUnavailable` would be more appropriate.

https://peps.python.org/pep-0008/#exception-names

It probably makes sense to fix this separately though, there might be other similar cases elsewhere.